.github/workflows/publish_latest_server_image.yml

```
@@ -0,0 +1,97 @@
+name: "Publish latest server docker image automatically"
+# todo: expand the yaml to deal with multi release version 
+on:
+  schedule:
+    - cron: '0 23 * * *'
+  workflow_dispatch:
+
+jobs:
+  build_latest:
+    runs-on: ubuntu-latest
+    env:
+      repository_url: apache/incubator-hugegraph
+      branch: master
+      image_url: hugegraph/hugegraph:latest
+
+    steps:
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+    - name: Checkout latest
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ env.repository_url }}
+        ref: ${{ env.branch }}
+        fetch-depth: 2
+
+    - name: Get current SHA
+      run: |
+        current_sha=$(git rev-parse HEAD)
+        echo "current_sha=$current_sha" >> $GITHUB_ENV
+
+    - name: Get last SHA
+      run: |
+        is_update='true'
```

rename to `need_update`, and default value better be `false` (and adaptor the logic below)

```suggestion
        need_update='true'
```